ext/docker/runmount_builder_test.go

```
@@ -0,0 +1,141 @@
+	_, cctl := ctl.CmdFor("docker", "create")
+	cctl.ResultSuccess(fmt.Sprintf("%s", testContainerBuildID), "")
+	ctx := sous.BuildContext{
+		Sh: sh,
+	}
+
+	containerID, err := createMountContainer(ctx, testBuildID)
+	assert.Empty(t, err)
+	assert.Equal(t, testContainerBuildID, containerID)
+}
+
+func getTestRunSpec() MultiImageRunSpec {
+	testRunSpec := MultiImageRunSpec{}
+	specF, _ := os.Open("./testdata/runmountbuilder/run_spec.json")
+	dec := json.NewDecoder(specF)
+	dec.Decode(&testRunSpec)
+
+	return testRunSpec
+}
+
+func TestRunmountBuild_extractRunspec(t *testing.T) {
+	tempDir := "./testdata/runmountbuilder"
+	testRunSpec := getTestRunSpec()
+	sh, ctl := shell.NewTestShell()
+	_, cctl := ctl.CmdFor("docker", "cp")
+	cctl.ResultSuccess("", "")
+	ctx := sous.BuildContext{
+		Sh: sh,
+	}
+
+	runSpec, _ := extractRunSpec(ctx, tempDir, testContainerBuildID)
+	fmt.Println("runspec : ", runSpec)
+
+	assert.Equal(t, testRunSpec, runSpec)
+}
+
+func TestRunmountBuild_validateRunSpec(t *testing.T) {
+	testRunSpec := getTestRunSpec()
+	err := validateRunSpec(testRunSpec)
+	assert.NoError(t, err)
+}
+
+func TestRunmountBuild_constructImageBuilders(t *testing.T) {
+	testRunSpec := getTestRunSpec()
+	builders, err := constructImageBuilders(testRunSpec)
+
+	assert.NoError(t, err)
+	builder := *builders[0]
+	assert.Equal(t, "docker.otenv.com/sous-otj-run", builder.RunSpec.Image.From)
+}
+
+func TestRunmountBuild_extractFiles(t *testing.T) {
+	testRunSpec := getTestRunSpec()
+	sh, ctl := shell.NewTestShell()
+	_, cctl := ctl.CmdFor("docker", "cp")
+	cctl.ResultSuccess("", "")
+	ctx := sous.BuildContext{
+		Sh: sh,
+	}
+	builders, _ := constructImageBuilders(testRunSpec)
+	err := extractFiles(ctx, testContainerBuildID, "/tmp", builders)
+	assert.NoError(t, err)
+}
+
+func TestRunmountBuild_templateDockerfile(t *testing.T) {
+	testRunSpec := getTestRunSpec()
+	buildDir := "/tmp"
+	builders, _ := constructImageBuilders(testRunSpec) //could abstract this for testing
```

OK, I'm sounding like a broken record at this point (maybe those other comments are below though) I don't think there are good reasons to ignore errors, even in tests. If it's making tests noisy, a good idea is to add a test helper func that wraps this, takes a `*testing.T` and always asserts the error is nil, and then call that instead. E.g.:

```
func mustConstructImageBuilders(t *testing.T, runspec MultiImageRunSpec) []*runnableBuilder {
    t.Helper()
    r, err := constructImageBuilders(runspec)
    if err != nil {
        t.Fatal(err)
    }
    return r
}
```

---

Just saw that this method always returns nil error, so the right thing is to remove that return value. The above may still be interesting though?